### PR TITLE
WIP: changed Battery A and Battery B references to a list

### PR DESCRIPTION
**Pull request by Kevinwochan**
https://github.com/ocius/api/issues/13

1. Battery data is available from the tqb XML entity. (https://github.com/ocius/api/blob/master/API%20Example%20XML/ExampleMavStatus.xml)
3. This is fetched and parsed into a JObject by the XmlToJson Lambda function and then inserted into the DynamoDB table.
4. The RawToClientData intercepts the insertion, extracts the battery data using the Battery model, assigns the BatteryA and BatteryB fields from the Battery voltage attribute and provides default values for unavailable data.
5. This is served by the GetAll API Gateway under the keys "BatteryA" and "BatteryB"

Changes
---
Added a ParseVoltage utility function that inserts the decimal point in the battery voltage string
All references to BatteryA and BatteryB were changed to a list of strings.

Integration requirements
---
Requires a new DynamoDB table with a schema allowing the battery field to be list
Requires changes to the front end to use the "batteries" key to access battery data

### CODE REVIEW

File: Services/RawDataToClientData/RawDataToClientData/Models/DroneUtils.cs [hunk 1]
```diff
@@ -11,8 +11,9 @@ public static string ParseDecimal(JToken token)
             return (decimalValue / offset).ToString();
         }
         public static string ParseBattery(Battery battery)
-        {
-            return battery.Vol.Insert(2, ".");
+            var voltage = battery.Vol;
```
Comment (Kevinwochan) [inline Services/RawDataToClientData/RawDataToClientData/Models/DroneUtils.cs:14]: I think this change might catch more cases? (assuming mav status is always going to provide values accurate up to 1 decimal place)
Currently, when mavstatus gives us "290 "to mean "29.0" inserting at index 2 works.
But when the voltage is "2900" which should mean "290.0", inserting at index 2 will create the string "29.00".

**Comment by Kevinwochan** [comment]
RawDataToClientData has been successfully deployed
image

**Comment by Kevinwochan** [comment]
GetAll endpoint has't been deployed yet.

**Comment by thomasdane** [comment]
> RawDataToClientData has been successfully deployed
> image

Nice work!

File: Services/RawDataToClientData/RawDataToClientData/Models/DroneSensors.cs [hunk 2]
```diff
@@ -19,12 +20,13 @@ public class DroneSensors
...
         public string BatteryB { get; set; }
+        public string Batteries { get; set; }
         public string Cameras { get; set; }
 
         public static string GetSensors(string name, string data, string cameras)
         {
-            var batteries = JsonConvert.DeserializeObject<Batteries>(data);
-
+            var batteryData = JsonConvert.DeserializeObject<Batteries>(data);
+            var batteries = batteryData.Tqb.Select(battery => DroneUtils.ParseVoltage(battery));
```
Comment (thomasdane) [inline Services/RawDataToClientData/RawDataToClientData/Models/DroneSensors.cs:29]: Nice use of select

File: Services/RawDataToClientData/RawDataToClientDataTests/DroneUtilsTests.cs [hunk 3]
```diff
@@ -20,6 +20,20 @@ public void ParseDecimal_WhenCoordinate_ReturnsLat()
             //Assert
             Assert.Equal(expected, actual);
         }
+
+        [Fact]
+        public void ParseBattery_WhenBattery_ReturnsVoltage()
```
Comment (thomasdane) [inline Services/RawDataToClientData/RawDataToClientDataTests/DroneUtilsTests.cs]: ❤️

**Pull request merged**